tests/neg/enum-values.scala

```diff
@@ -0,0 +1,25 @@
+enum Tag[T]:
+  case Int extends Tag[Int]
+  case String extends Tag[String]
+  case OfClass[T]()(using val tag: reflect.ClassTag[T]) extends Tag[T]
+
+enum ListLike[+T]:
+  case Cons[T](head: T, tail: ListLike[T]) extends ListLike[T]
+  case EmptyListLike
+
+enum TypeCtorsK[F[_]]:
+  case List       extends TypeCtorsK[List]
+  case Option     extends TypeCtorsK[Option]
+  case Const[T]() extends TypeCtorsK[[U] =>> T]
+
+def Test: Unit =
+  import Tag._, ListLike._, TypeCtorsK._
+  import reflect.Selectable.reflectiveSelectable
+
+  val tags: Array[Tag[?]]              = Tag.values // error: value values is not a member of object Tag
+  val listlikes: Array[ListLike[?]]    = ListLike.values // error: value values is not a member of object ListLike
+  val typeCtorsK: Array[TypeCtorsK[?]] = TypeCtorsK.values // error: value values is not a member of object TypeCtorsK
```

Comment:
At the SIP meeting, Josh pointed out that it would be good to make an extra effort for good error messages for those. Something that says "although Tag is an enum, it contains non-singleton cases, which prevents it from having a `values` array".

Same for `valueOf`.

How hard would be to do that?

Reply:
I have a version using @compileTimeOnly, rather than special casing missing member

Reply:
although now I see it that probably is not correct to use as it still makes byte code

Reply:
I can use compiletime.error in an inline def